src/browser/provider/built-in-providers.js

```diff
@@ -0,0 +1,137 @@
+    resizeCorrections: {},
+
+    useResizeHack: OS.win,
+
+    async _calculateResizeCorrections (id) {
+        await this.waitForConnectionReady(id);
+
+        var { width, height, title } = await this.runInitScript(id, getDataFromBrowser.toString());
+
+        if (!await browserNatives.isMaximized(title))
+            return;
+
+        await browserNatives.resize(title, width, height, width, height);
+
+        var { width: newWidth, height: newHeight } = await this.runInitScript(id, getDataFromBrowser.toString());
+
+        await browserNatives.maximize(title);
+
+        if (newWidth === width && newHeight === height)
+            return;
+
+        this.resizeCorrections[id] = { width: newWidth - width, height: newHeight - height };
+    },
+
+
+    async openBrowser (id) {
+        if (this.useResizeHack)
+            await this._calculateResizeCorrections(id);
+    },
+
+    async resizeWindow (id, pageInfo, width, height) {
+        // TODO: remove once https://github.com/DevExpress/testcafe-browser-natives/issues/12 implemented
+        if (OS.linux) {
+            this.reportWarning(id, WARNING_MESSAGE.browserManipulationsNotSupportedOnLinux);
+            return;
+        }
+
+        if (this.resizeCorrections[id]) {
+            width -= this.resizeCorrections[id].width;
+            height -= this.resizeCorrections[id].height;
+
+            delete this.resizeCorrections[id];
+        }
+
+        await browserNatives.resize(pageInfo.title, pageInfo.width, pageInfo.height, width, height);
+    },
+
+    async screenshotWindow (id, pageInfo, screenshotPath) {
+        // TODO: remove once https://github.com/DevExpress/testcafe-browser-natives/issues/12 implemented
+        if (OS.linux) {
+            this.reportWarning(id, WARNING_MESSAGE.browserManipulationsNotSupportedOnLinux);
+            return;
+        }
+
+        await browserNatives.screenshot(pageInfo.title, screenshotPath);
+    }
+};
+
+export var PathBrowserProvider = assignIn({}, BrowserProviderBase, {
+    async openBrowser (id, descriptor, startPage) {
+        await browserNatives.open(descriptor, startPage);
+
+        await BrowserProviderBase.openBrowser.call(this, id, descriptor, startPage);
+    },
+
+    async closeBrowser (id, pageInfo) {
+        await browserNatives.close(pageInfo.title);
+    },
+
+    async getDescriptorDefaultValues (alias) {
+        return browserNatives.getBrowserInfo(alias);
+    }
+});
+
+export var LocalBrowserProvider = assignIn({}, PathBrowserProvider, {
+    async listAvailableAliases () {
+        return Object.keys(await browserNatives.getInstallations());
+    },
+
+    async isValidAlias (alias) {
+        return (await this.listAvailableAliases()).indexOf(alias) !== -1;
+    }
+});
+
+export var RemoteBrowserProvider = assignIn({}, BrowserProviderBase, {
+    async openBrowser (id, descriptor, startPage) {
+        try {
+            await BrowserProviderBase.openBrowser.call(this, id, descriptor, startPage);
```

Comment:
Since we have inheritance here, it makes sense to implement these providers as classes and export newly created instances. Also, maybe it makes sense to move them to separate files?